### PR TITLE
Support file scan a directory of parquet files for S3

### LINKED ISSUE #2408: Document: feat(batch): support file scan a directory of parquet files
This issue tracks the documentation update needed for the merged PR #17811.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17811
Source PR Merged At: 2024-07-25T11:07:52Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/17811
- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2408

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

File: docs/ingest/ingest-from-s3.md [hunk 1]
```diff
@@ -240,6 +246,57 @@ SELECT * FROM file_scan(
...
+  "row_count": 10000,
+  "column_count": 5,
+  "schema": {
+    "column1": "string",
+    "column2": "int64",
+    "column3": "double",
+    "column4": "boolean",
+    "column5": "timestamp"
+  }
+},
```
Comment (chenzl25) [inline docs/ingest/ingest-from-s3.md]: The result format of reading a directory of parquets should be the same as reading one parquet file but we merge the results together

**Review by chenzl25 — APPROVED** [review]
LGTM

File: docs/ingest/ingest-from-s3.md [hunk 2]
```diff
@@ -216,30 +216,82 @@ CREATE MATERIALIZED VIEW mv AS SELECT * FROM s3_source;
...
 
 ```sql title="Function signature"
-file_scan(file_format, storage_type, s3_region, s3_access_key, s3_secret_key, file_location)
+file_scan(file_format, storage_type, s3_region, s3_access_key, s3_secret_key, file_location_or_directory)
 ```
 
-```sql title="Examples"
-SELECT * FROM file_scan(
+:::note
+Note: When reading a directory of Parquet files, the schema will be based on the first Parquet file listed. Please ensure that all Parquet files in the directory have the same schema.
```
Comment (WanYixian) [inline docs/ingest/ingest-from-s3.md]:
```suggestion
When reading a directory of Parquet files, the schema will be based on the first Parquet file listed. Please ensure that all Parquet files in the directory have the same schema.
```

**Review by WanYixian — APPROVED** [review]
LGTM

**Review by xxchan — COMMENTED** [review]
I think the organization is problematic. The top level "Syntax" is for CREATE s3 SOURCE. `file_scan` should be parallel to the source, but not its child.

image

**Pull request merged**